client/tests/test_client.py

```diff
@@ -17,38 +17,12 @@ def test_delete_client(self, saved_client0):
         assert saved_client0 not in Client.objects.all()
 
     @pytest.mark.django_db()
-    def test_delete_not_exist_clinet(self, client0):
-        assert client0 not in Client.objects.all()
-        with pytest.raises(ObjectDoesNotExist):
-            client0.delete_client()
-
-    @pytest.mark.django_db()
-    def test_fail_to_add_client_user_name(self, client0):
-        short_user_name = 'lior'
-        client0.user_name = short_user_name
-        with pytest.raises(ValidationError):
-            client0.save_client()
-
-    @pytest.mark.django_db()
-    def test_fail_to_add_client_first_name(self, client0):
-        empty_field = ''
-        client0.first_name = empty_field
-        with pytest.raises(ValidationError):
-            client0.save_client()
-
-    @pytest.mark.django_db()
-    def test_fail_to_add_client_last_name(self, client0):
-        empty_field = ''
-        client0.last_name = empty_field
-        with pytest.raises(ValidationError):
-            client0.save_client()
-
-    @pytest.mark.django_db()
-    def test_fail_to_add_client_password(self, client0):
-        short_password = '123'
-        client0.password = short_password
-        with pytest.raises(ValidationError):
-            client0.save_client()
+    def test_delete_not_exist_clinet(self, saved_client0):
+        with pytest.raises(Exception):
+            assert saved_client0 in Client.objects.all()
+            saved_client0.delete_client()
+            assert saved_client0 not in Client.objects.all()
+            saved_client0.delete_client()
```

Comment:
Which line is supposed to generate the exception? Have only that line inside the context manager. You won't reach any line below it and those assertions will never be reached.

Reply:
line 25 should raise the exception, in the delete method, get method is used.

Reply:
Alright. The code should be changed so that line is the only line inside the context manager.

Reply:
but if we delete straight away non existing client from the db, how do we prove that he there before? yea of course there will be an exception raised but dont we want to prove it was there before in the db?

Reply:
I'm not saying take it off the test. I'm saying take it off the context manager. i.e.
```suggestion
        assert saved_client0 in Client.objects.all()
        saved_client0.delete_client()
        assert saved_client0 not in Client.objects.all()
        with pytest.raises(Exception):
            saved_client0.delete_client()
```

Reply:
ohhhhh... ok yea i agree